### PR TITLE
fix issue#2497

### PR DESCRIPTION
**Pull request by broccoliSpicy**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2497 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/vectorize/bin/bin_test.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ func TestCountBitLenForInt(t *testing.T) {
 	require.Equal(t, int64(64+64+24+30), Int64BitLen([]int64{-1e7, -1e9, 1e7, 1e9}))
 }
 
+// due to differences between x86/arm, this TestCountBitLenForFloat function has some compatibility issues and therefore commented out
+/*
```
Comment (fengttt) [inline pkg/vectorize/bin/bin_test.go:47]: Why do you ever want to count bit in a float?    I would completely remove all related code.

Comment (broccoliSpicy) [reply pkg/vectorize/bin/bin_test.go:47]:
```go
func bitLen[T constraints.Integer | constraints.Float](xs []T) int64 {
	var r int64
	for _, x := range xs {
		// convert to uint64 to count bits
		val := uint64(x)
		if val == 0 {
			// special case for zero
			r += 1
		} else {
			r += int64(bits.Len64(val))
		}
	}
	return r
}
```

```sql
mysql> create table i_f(a bigint unsigned, b double);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into i_f values(23, 23.4);
Query OK, 1 row affected (0.00 sec)

mysql> select bin(a), bin(b) from i_f;
+--------+--------+
| bin(a) | bin(b) |
+--------+--------+
| 10111  | 10111  |
+--------+--------+
1 row in set (0.00 sec)
```

for float values, perhaps the semantic for this function here is to return the binary representation of the uint64 casted value of the float, @e11jah

Comment (broccoliSpicy) [reply pkg/vectorize/bin/bin_test.go:47]: the compatibility issue is caused by math/bits bits.Len64 function
```go
package main

import "fmt"
import "math/bits"
import "golang.org/x/exp/constraints"
import . "github.com/klauspost/cpuid/v2"
// import "golang.org/x/sys/cpu"

func bitLen[T constraints.Integer | constraints.Float](xs []T) int64 {
    var r int64
    for _, x := range xs {
        // convert to uint64 to count bits
        val := uint64(x)
        if val == 0 {
            // special case for zero
            r += 1
        } else {
            r += int64(bits.Len64(val))
        }
    }
    return r
}

func main() {
    fmt.Println(CPU.BrandName)
    fmt.Println("bitLen function result ", bitLen([]float32{-1.99}))
}
```

amd64 result:
Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
bitLen function result  64

arm64 result:

bitLen function result  1

Comment (broccoliSpicy) [reply pkg/vectorize/bin/bin_test.go:47]: no, I think I was wrong, math/bits bitLen is fine here, 
in Golang, cast operations like this is illegal:
```go
func main() {
    fmt.Println(uint64(float32(-1.99)))
}
```
```
➜  playground go run play.go
# command-line-arguments
./play.go:40:24: cannot convert float32(-1.99) (constant -1.99 of type float32) to type uint64
```

```go
package main

import "fmt"
import "math/bits"
import "golang.org/x/exp/constraints"
import . "github.com/klauspost/cpuid/v2"
// import "golang.org/x/sys/cpu"

func bitLen[T constraints.Integer | constraints.Float](xs []T) int64 {
    var r int64
    for _, x := range xs {
        // convert to uint64 to count bits
        val := uint64(x)
        fmt.Println(val)
        if val == 0 {
            // special case for zero
            r += 1
        } else {
            r += int64(bits.Len64(val))
        }
    }
    return r
}

func main() {
    fmt.Println(CPU.BrandName)
    fmt.Println("bitLen function result ", bitLen([]float32{-1.99}))
}
```

amd64 result:
```
Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
18446744073709551615
bitLen function result  64
```

arm64 result:
```

0
bitLen function result  1
```

for some kind of reasons I don't know, Golang didn't threw out a error here, and arm64 returns 0 while amd64 returns 18446744073709551615(2^64 − 1), the maximum value of uint64
@e11jah

Comment (broccoliSpicy) [reply pkg/vectorize/bin/bin_test.go:47]: is it possible that our implementation for function bin has mistakes when the type is float? @e11jah

Comment (e11jah) [reply pkg/vectorize/bin/bin_test.go:47]: The behavior of casting a negative float to an unsigned int is defined by golang compiler. This issue also exists in c and c++. But the behavior is the same on x86 and arm64 in rust.😂

https://embeddeduse.com/2013/08/25/casting-a-negative-float-to-an-unsigned-int/

**Pull request merged**